src/use-swr.ts

```diff
@@ -157,7 +158,7 @@ function useSWR<Data = any, Error = any>(
   const shouldReadCache = config.suspense || !useHydration()
 
   // stale: get from cache
-  let [data, setData] = useState(shouldReadCache ? cacheGet(key) : undefined)
+  let [data = config.initialData, setData] = useState((shouldReadCache ? cacheGet(key) : undefined) || config.initialData)
```

Comment:
So, without this `data = config.initialData`, the very first render doesn't have the correct data in it. This definitely has a code-smell, but I'm not sure how to do it better?

Reply:
Yeah normally the first render won't have cache available, so it will always be undefined. I think the solution will be either you provide `initialData` (e.g. from getInitialProps if you're using SSR) or use suspense.